src/azul/chalice.py

```diff
@@ -145,3 +153,107 @@ def _log_response(self, response):
     # Some type annotations to help with auto-complete
     lambda_context: LambdaContext
     current_request: Request
+
+
+class ValidatingAzulChaliceApp(AzulChaliceApp):
+    """
+    This class allows for a validation method to decorate the called view
+    function.
+    """
+
+    def __init__(self, app_name, unit_test=False, spec=None):
+        super().__init__(app_name=app_name, unit_test=unit_test, spec=spec)
+
+    # override so we can call _validate
+    def route(self, *args, validate: bool = True, **kwargs):
+        """
+        See docstring from super class for additional information.
+        :param validate: If False, do not validate requests submitted to the
+        endpoint.
+        """
+
+        azul_chalice_decorator = super().route(*args, **kwargs)
+        if validate:
+            def decorator(view_func):
+                @functools.wraps(view_func)
+                def f(*args, **kwargs):
+                    status = self.validate_params(self.current_request)
+                    if isinstance(status, Response):
+                        return status
+                    return view_func(*args, **kwargs)
+
+                return azul_chalice_decorator(f)
+
+            return decorator
+        else:
+            return azul_chalice_decorator
+
+    def get_spec_validators(self, path, http_method) -> Dict[str, SpecValidator]:
+        spec_validators = dict()
+        for parameter_spec in [
+            *self.specs['paths'][path].get('parameters', []),
```

Comment:
use empty tuple for default. It's should be cheaper. Tuples are immutable so they can be shared. In fact, there is only one empty tuple instance in Python. But most times you write `[]`, a new mutable list is allocated on the heap.

```
id([])
140129130225216
id([])
140129130206336
id([])
140129130206336
id([])
140129130206784
id([])
140129130206336
```

There seems to be some reuse going on. But the empty tuple is always the same

```
id(())
140129154482240
id(())
140129154482240
id(())
140129154482240
id(())
140129154482240
id(())
140129154482240
```